src/js/lesson_1/homework.js

```diff
@@ -30,7 +47,15 @@ function factorial(number) {
  * @return {string} entered name
  */
 function greetAndConfirm() {
-  // ...
+  let checking;
+  do {
+    var name = prompt('Enter your name');
+    if (name === '') {
```

Comment:
Я бы предложил использовать 
```suggestion
    if (!name) {
```